db/system_distributed_keyspace.cc

```diff
@@ -582,8 +584,10 @@ future<qos::service_levels_info> system_distributed_keyspace::get_service_levels
         qos::service_levels_info service_levels;
         for (auto &&row : *result_set) {
             auto service_level_name = row.get_as<sstring>("service_level");
+            auto workload = qos::service_level_options::parse_workload_type(row.get_opt<sstring>("workload_type").value_or(""));
```

Comment:
This will throw if the value is wrong, killing the node. Better to ignore (with a warning) illegal values here.

Reply:
Also, two more issues regarding this piece of code:
 - killing the node is a preexisting condition, since we already parse service level name and the timeout duration in the same way
 - it's actually redundant to get the service level name from the row - it's already passed as a parameter to this function

I'll just add a commit which fixes both issues to this series

Reply:
Doesn't parse_workload_type throw?

Reply:
Nope, it returns an optional. I guess I spent too much time with Rust to produce throwing code.